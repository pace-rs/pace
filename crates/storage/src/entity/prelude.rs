//! `SeaORM` Entity. Generated by sea-orm-codegen 0.12.15

pub use super::activities::Entity as Activities;
pub use super::activities_categories::Entity as ActivitiesCategories;
pub use super::activities_tags::Entity as ActivitiesTags;
pub use super::activity_kinds::Entity as ActivityKinds;
pub use super::activity_status::Entity as ActivityStatus;
pub use super::categories::Entity as Categories;
pub use super::descriptions::Entity as Descriptions;
pub use super::tags::Entity as Tags;
