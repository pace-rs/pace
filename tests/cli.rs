use assert_cmd::Command;
// use pace_core::ActivityLog;
use predicates::prelude::predicate;
// use tempfile::tempdir;

pub type TestResult<T> = Result<T, Box<dyn std::error::Error>>;

pub fn pace_runner(/*temp_dir: &TempDir*/) -> TestResult<Command> {
    // TODO: when we have implemented init, we can use this to create a new pace project
    // let _repo_dir = temp_dir.path();

    let runner = Command::new(env!("CARGO_BIN_EXE_pace"));

    Ok(runner)
}

// TODO: when we have implemented init, we can use this to create a new pace project
// fn setup() -> TestResult<TempDir> {
//     let temp_dir = tempdir()?;
//     pace_runner(&temp_dir)?.args(["init"]).assert().success();

//     Ok(temp_dir)
// }

#[test]
fn test_version_command_passes() -> TestResult<()> {
    _ = pace_runner()?
        .arg("--version")
        .assert()
        .success()
        .stdout(predicate::str::contains(env!("CARGO_PKG_VERSION")));

    Ok(())
}

#[test]
fn test_help_command_passes() -> TestResult<()> {
    _ = pace_runner()?
        .arg("--help")
        .assert()
        .success()
        .stdout(predicate::str::contains("Usage:"));

    Ok(())
}

// TODO!: Test begin command
// #[test]
// fn test_begin_command_passes() -> TestResult<()> {
//     let activity_log_file = tempdir()?.into_path().join("activity_log.toml");

//     let desc = "Test description";
//     let category = "Test::Category";
//     let time = "22:00";

//     if activity_log_file.exists() {
//         std::fs::remove_file(&activity_log_file)?;
//     }

//     std::fs::write(&activity_log_file, "")?;

//     _ = pace_runner()?
//         .args([
//             "-a",
//             activity_log_file.as_path().to_str().unwrap(),
//             "begin",
//             desc,
//             "-c",
//             category,
//             "-t",
//             time,
//         ])
//         .assert()
//         .success()
//         .stdout(predicate::str::contains("started"));

//     let contents = std::fs::read_to_string(&activity_log_file)?;

//     let activity_log = toml::from_str::<ActivityLog>(&contents)?;

//     assert_eq!(activity_log.activities().len(), 1);

//     let activity = activity_log.activities().front().unwrap();

//     assert_eq!(activity.description(), &Some(desc.to_string()));
//     assert_eq!(activity.category(), &Some(category.to_string()));
//     assert_eq!(format!("{:?}", activity.begin()), time);

//     if activity_log_file.exists() {
//         std::fs::remove_file(&activity_log_file)?;
//     }

//     Ok(())
// }

// TODO: Test end command
// #[test]
// fn test_end_command_passes() -> TestResult<()> {
//     pace_runner()?
//         .args(["-a", "./activity_log.toml", "end"])
//         .assert()
//         .success()
//         .stdout(predicate::str::contains("finished")); // TODO

//     Ok(())
// }

// TODO: Test now command
// #[test]
// fn test_now_command_passes() -> TestResult<()> {
//     pace_runner()?
//         .args(["-a", "./activity_log.toml", "now"])
//         .assert()
//         .success()
//         .stdout(predicate::str::contains("current")); // TODO

//     Ok(())
// }
