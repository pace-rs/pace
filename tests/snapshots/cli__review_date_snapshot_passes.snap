---
source: tests/cli.rs
info:
  program: pace
  args:
    - "--config"
    - tests/fixtures/configs/pace.toml
    - "--activity-log-file"
    - "./tests/fixtures/activity_tracker/activities.pace.toml"
    - review
    - "--date"
    - 2024-02-26
---
success: true
exit_code: 0
----- stdout -----
╭─────────────┬─────────────────────┬─────────────────────┬─────────────────╮
│                 Your activity insights for the period:                    │
│                                                                           │
│                 2024-02-26 00:00:00 - 2024-02-26 23:59:59                 │
├─────────────┼─────────────────────┼─────────────────────┼─────────────────┤
│  Category   │     Description     │ Duration (Sessions) │ Breaks (Amount) │
├─────────────┼─────────────────────┼─────────────────────┼─────────────────┤
│ development │                     │              3m 22s │           4m 4s │
├─────────────┼─────────────────────┼─────────────────────┼─────────────────┤
│    pace     │ Implementing Review │          3m 22s (1) │       4m 4s (1) │
├─────────────┼─────────────────────┼─────────────────────┼─────────────────┤
│    Total    │                     │              3m 22s │           4m 4s │
╰─────────────┴─────────────────────┴─────────────────────┴─────────────────╯

----- stderr -----
